Controller/Process/Result.php

```diff
@@ -279,31 +279,37 @@ protected function _validateUpdateOrder($order, $response)
      */
     protected function _authenticate($response) {
 
-        $merchantSigNotification = $response['merchantSig'];
-
-        // do it like this because $_GET is converting dot to underscore
-        $queryString = $_SERVER['QUERY_STRING'];
-        $result = [];
-        $pairs = explode("&", $queryString);
-
-        foreach ($pairs as $pair) {
-            $nv = explode("=", $pair);
-            $name = urldecode($nv[0]);
-            $value = urldecode($nv[1]);
-            $result[$name] = $value;
-        }
+    	if (!empty($response['merchantSig'])) {
```

Comment:
Why not using different urls when in checkout and when in hpp?
This way we can have different functions and clearer method signatures instead of if-else statements.

Reply:
Hopefully we are going to depricate HPP in the near future so I tried to solve it with the less effort it can be. :)